includes/UserMailer.php

```diff
@@ -550,126 +583,104 @@ public function notifyOnPageChange( $editor, $title, $timestamp, $summary, $mino
 	 * Send emails corresponding to the user $editor editing the page $title.
 	 * Also updates wl_notificationtimestamp.
 	 *
-	 * @param $editor User object
-	 * @param $title Title object
-	 * @param $timestamp string Edit timestamp
-	 * @param $summary string Edit summary
-	 * @param $minorEdit bool
-	 * @param $oldid int Revision ID
-	 * @param $watchers array of user IDs
-	 * @param $action (Wikia)
-	 * @param $otherParam
 	 */
-	public function actuallyNotifyOnPageChange( $editor, $title, $timestamp, $summary, $minorEdit, $oldid, $watchers, $action='', $otherParam = array() ) {
-		# we use $wgPasswordSender as sender's address
+	private function actuallyNotifyOnPageChange( $watchers ) {
 		global $wgEnotifWatchlist;
 		global $wgEnotifMinorEdits, $wgEnotifUserTalk;
 
-		wfProfileIn( __METHOD__ );
+		$this->setReplyToAndFromAddresses();
 
 		# The following code is only run, if several conditions are met:
 		# 1. EmailNotification for pages (other than user_talk pages) must be enabled
 		# 2. minor edits (changes) are only regarded if the global flag indicates so
+		if ( !$this->minorEdit || ( $wgEnotifMinorEdits && !$this->editor->isAllowed( 'nominornewtalk' ) ) ) {
 
-		$isUserTalkPage = ( $title->getNamespace() == NS_USER_TALK );
-
-		$this->title = $title;
-		$this->timestamp = $timestamp;
-		$this->summary = $summary;
-		$this->minorEdit = $minorEdit;
-		$this->oldid = $oldid;
-		$this->action = $action;
-		$this->editor = $editor;
-		$this->composed_common = false;
-		$this->other_param = $otherParam;
-		
-		$userTalkId = false;
-
-		if ( !$minorEdit || ( $wgEnotifMinorEdits && !$editor->isAllowed( 'nominornewtalk' ) ) ) {
-
-			if ( $wgEnotifUserTalk && $isUserTalkPage && $this->canSendUserTalkEmail( $editor, $title, $minorEdit ) ) {
-				$targetUser = User::newFromName( $title->getText() );
+			$userTalkId = 0;
+			if ( $wgEnotifUserTalk && $this->isUserTalkPage() && $this->canSendUserTalkEmail() ) {
```

Comment:
Same with this conditional, can go into a function and also used in `shouldSendEmail()`